samples/snippets/src/main/java/pubsub/CreateProtoSchemaExample.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pubsub;
+
+// [START pubsub_create_proto_schema]
+
+import com.google.api.gax.rpc.AlreadyExistsException;
+import com.google.cloud.pubsub.v1.SchemaServiceClient;
+import com.google.pubsub.v1.ProjectName;
+import com.google.pubsub.v1.Schema;
+import com.google.pubsub.v1.SchemaName;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class CreateProtoSchemaExample {
+
+  public static void main(String... args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String schemaId = "your-schema-id";
+    String protoFile = "path/to/a/proto/file/(.proto)/formatted/in/protocol/buffers";
+
+    createProtoSchemaExample(projectId, schemaId, protoFile);
+  }
+
+  public static void createProtoSchemaExample(String projectId, String schemaId,
+      String protoFile)
+      throws IOException {
+
+    ProjectName projectName = ProjectName.of(projectId);
+    SchemaName schemaName = SchemaName.of(projectId, schemaId);
+
+    // Read a proto file as a string.
+    String protoSource = new String(Files.readAllBytes(Paths.get(protoFile)));
```

Comment:
@kamalaboulhosn Do you know a better way than just reading the file as a string? The equivalent Avro example uses a parser() that also validates the Avro schema string.

Reply:
Hmmm...I'm not sure it's really necessary to validate the proto in the client. The server validates the proto, so parsing it on the client probably isn't necessary. There isn't a way to parse the proto file in Java. This is the reason we can't support proto in the emulator.